test/functional/test_framework/wallet.py

```diff
@@ -55,25 +55,34 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), seq_num = 0xffffffff, from_node, utxo_to_spend=None, send_tx = True, default_fee = False):
         """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
         self._utxos = sorted(self._utxos, key=lambda k: k['value'])
         utxo_to_spend = utxo_to_spend or self._utxos.pop()  # Pick the largest utxo (if none provided) and hope it covers the fee
         vsize = Decimal(96)
-        send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000))
+
+        # if send_tx is set to False then calculate send_value without vsize
+        if send_tx:
+            send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000))
+        else:
+            send_value = satoshi_round(utxo_to_spend['value'] - fee_rate)
         fee = utxo_to_spend['value'] - send_value
         assert send_value > 0
 
         tx = CTransaction()
-        tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']))]
+        tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']), nSequence = seq_num)]
         tx.vout = [CTxOut(int(send_value * COIN), self._scriptPubKey)]
         tx.wit.vtxinwit = [CTxInWitness()]
         tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
         self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
```

Comment:
If `send_tx` was set to `False`, this utxo could be unspendable, potentially causing problems downstream where the `get_utxo` does not yield a spendable utxo. So I it's probably better to only append the new utxo if `send_tx == True`, after calling `sendrawtransaction`.

Reply:
Thank you for the suggestion! I moved the append utxo after `sendrawtransaction`